.evergreen/config.yml

```diff
@@ -296,7 +296,7 @@ functions:
         working_dir: "mongoc"
         script: |
           set -o errexit
-          COMPRESSORS="${COMPRESSORS}" CC="${CC}" AUTH=${AUTH} SSL=${SSL} URI=${URI} IPV4_ONLY=${IPV4_ONLY} VALGRIND=${VALGRIND} sh .evergreen/run-tests.sh
+          COMPRESSORS="${COMPRESSORS}" CC="${CC}" AUTH=${AUTH} SSL=${SSL} URI=${URI} IPV4_ONLY=${IPV4_ONLY} VALGRIND=${VALGRIND} MONGOC_TEST_URI=${URI} sh .evergreen/run-tests.sh
```

Comment:
All the existing tests in config.yml specify URI="", which is good because the URI would have no effect even if it were non-empty. This change makes the tests actually use a non-empty URI.

Later, I'll delete all those URI="" lines, because specifying an empty URI *still* has no effect.